stdlib/builtins.pyi

```diff
@@ -898,8 +898,21 @@ class memoryview(Sequence[int]):
     def __buffer__(self, __flags: int) -> memoryview: ...
     def __release_buffer__(self, __buffer: memoryview) -> None: ...
 
+class _Truthy(Protocol):
+    def __bool__(self) -> Literal[True]: ...
+
+class _Falsy(Protocol):
+    def __bool__(self) -> Literal[False]: ...
+
 @final
 class bool(int):
+    @overload
+    def __new__(cls) -> Literal[False]: ...
+    @overload
+    def __new__(cls, __o: _Truthy) -> Literal[True]: ...
+    @overload
+    def __new__(cls, __o: _Falsy) -> Literal[False]: ...
+    @overload
     def __new__(cls, __o: object = ...) -> Self: ...
```

Comment:
Slightly simpler:

```suggestion
    @overload
    def __new__(cls, __o: _Truthy) -> Literal[True]: ...
    @overload
    def __new__(cls, __o: _Falsy = ...) -> Literal[False]: ...
    @overload
    def __new__(cls, __o: object) -> Self: ...
```

Reply:
pyright's not a fan <img width="764" alt="Screenshot 2023-07-19 at 11 47 32" src="https://github.com/python/typeshed/assets/50501825/5807da25-41fb-47f0-ad3f-79f75d43c92d">

Reply:
I think pyright is incorrect (and we have that lint switched off in our CI due to the number of false positives it caused). But if you like, you can add some test cases to prove me wrong :)